pennylane/ops/identity.py

```diff
@@ -177,3 +177,167 @@ def adjoint(self):
+
+    @staticmethod
+    def compute_eigvals(phi, n_wires=1):  # pylint: disable=arguments-differ
+        r"""Eigenvalues of the operator in the computational basis (static method).
+
+        If :attr:`diagonalizing_gates` are specified and implement a unitary :math:`U^{\dagger}`,
+        the operator can be reconstructed as
+
+        .. math:: O = U \Sigma U^{\dagger},
+
+        where :math:`\Sigma` is the diagonal matrix containing the eigenvalues.
+
+        Otherwise, no particular order for the eigenvalues is guaranteed.
+
+        .. seealso:: :meth:`~.GlobalPhase.eigvals`
+
+        Returns:
+            array: eigenvalues
+
+        **Example**
+
+        >>> qml.GlobalPhase.compute_eigvals(np.pi/2)
+        array([6.123234e-17+1.j, 6.123234e-17+1.j])
+        """
+        return qml.math.exp(-1j * phi) * qml.math.ones(2**n_wires)
+
+    @staticmethod
+    def compute_matrix(phi, n_wires=1):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.GlobalPhase.matrix`
+
+        Returns:
+            ndarray: matrix
+
+        **Example**
+
+        >>> qml.GlobalPhase.compute_matrix(np.pi/4, n_wires=1)
+        array([[0.70710678-0.70710678j, 0.        +0.j        ],
+               [0.        +0.j        , 0.70710678-0.70710678j]])
+        """
+        return qml.math.exp(-1j * qml.math.cast(phi, complex)) * qml.math.eye(int(2**n_wires))
+
+    @staticmethod
+    def compute_sparse_matrix(phi, n_wires=1):  # pylint: disable=arguments-differ
+        return qml.math.exp(-1j * phi) * sparse.eye(int(2**n_wires), format="csr")
+
+    @staticmethod
+    def compute_diagonalizing_gates(
+        phi, wires, n_wires=1
+    ):  # pylint: disable=arguments-differ,unused-argument
+        r"""Sequence of gates that diagonalize the operator in the computational basis (static method).
+
+        Given the eigendecomposition :math:`O = U \Sigma U^{\dagger}` where
+        :math:`\Sigma` is a diagonal matrix containing the eigenvalues,
+        the sequence of diagonalizing gates implements the unitary :math:`U^{\dagger}`.
+
+        The diagonalizing gates rotate the state into the eigenbasis
+        of the operator.
+
+        .. seealso:: :meth:`~.GlobalPhase.diagonalizing_gates`.
+
+        Args:
+            wires (Iterable[Any], Wires): wires that the operator acts on
+
+        Returns:
+            list[.Operator]: list of diagonalizing gates
+
+        **Example**
+
+        >>> qml.GlobalPhase.compute_diagonalizing_gates(1.2, wires=[0])
+        []
+        """
+        return []
+
+    def matrix(self, wire_order=None):
+        n_wires = len(wire_order) if wire_order else len(self.wires)
```

Comment:
`len(self.wires)` will always be 0, right? not sure if we want the default to be 0 or 1 for the matrix size.

Reply:
Good point, will default to 1

Reply:
It now does take wires, and if you pass it wires, it gives you a matrix with the appropriate dimensions for the wires you passed. Otherwise things break. So now it defaults to `np.max([1, len(self.wires)])`.

Reply:
now it defaults to `len(self.wires)` again because `apply_operation` seems to assume it will